pkg: fix test that would fail on non-amd64

gridbugs opened the pull request.

gridbugs committed:
pkg: fix test that would fail on non-amd64

Signed-off-by: Stephen Sherratt <stephen@sherra.tt>

Leonidas-from-XIV commented:
Can you explain what the failure is? I can't test the current code because my system is amd64 (and apparently all of our CI as well).

gridbugs commented:
On arm64 systems the `arm64-only` dependency was chosen:
```diff
File "test/blackbox-tests/test-cases/pkg/lock-directory-selection.t", line 1, characters 0-0:
diff --git a/_build/.sandbox/0d0b752274066769dfcc6df483807154/default/test/blackbox-tests/test-cases/pkg/lock-directory-selection.t b/_build/.sandbox/0d0b752274066769dfcc6df483807154/default/test/blackbox-tests/test-cases/pkg/lock-directory-selection.t.corrected
index 3279630f2..d414d608d 100644
--- a/_build/.sandbox/0d0b752274066769dfcc6df483807154/default/test/blackbox-tests/test-cases/pkg/lock-directory-selection.t
+++ b/_build/.sandbox/0d0b752274066769dfcc6df483807154/default/test/blackbox-tests/test-cases/pkg/lock-directory-selection.t.corrected
@@ -133,7 +133,9 @@ Test that cond statements can have a default value:

   $ dune pkg lock dune.lock
   Solution for dune.lock:
+  - arm64-only.0.0.1
   - linux-only.0.0.1
   $ dune clean
   $ dune build _build/_private/default/.pkg/linux-only/target/
+  arm64-only
   linux-only

```
